purchase_request_to_rfq/wizard/purchase_request_line_make_purchase_order.py

```diff
@@ -138,37 +135,40 @@ def _prepare_purchase_order_line(self, po, item):
         po_line_obj = self.env['purchase.order.line']
         product = item.product_id
         supplier = self.supplier_id
-        pricelist_id = supplier.property_product_pricelist
-
-        if pricelist_id:
-            date_order_str = datetime.strptime(
-                fields.datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT),
-                DEFAULT_SERVER_DATETIME_FORMAT).\
-                strftime(DEFAULT_SERVER_DATE_FORMAT)
-            price = pricelist_id.price_get(prod_id=product.id,
-                                           qty=item.product_qty or 1.0,
-                                           partner=supplier or False,
-                                           context={'uom':
-                                                    product.uom_po_id.id,
-                                                    'date': date_order_str})
-            price = price[pricelist_id.id]
-        else:
-            price = product.standard_price
-
+        taxes = item.product_id.supplier_taxes_id
+        fpos = po.fiscal_position_id
+        taxes_id = fpos.map_tax(taxes) if fpos else taxes
+        if taxes_id:
+            taxes_id = taxes_id.filtered(
+                lambda x: x.company_id.id == po.company_id.id)
+        procurement_uom_po_qty = self.env['product.uom']._compute_qty_obj(
+            item.product_uom_id, item.product_qty, item.product_id.uom_po_id)
+        seller = item.product_id._select_seller(
+            item.product_id,
+            partner_id=supplier,
+            quantity=procurement_uom_po_qty,
+            date=po.date_order and po.date_order[:10],
+            uom_id=item.product_id.uom_po_id)
+        price_unit = self.env['account.tax']._fix_tax_included_price(
+            seller.price, item.product_id.supplier_taxes_id, taxes_id) \
+            if seller else 0.0
+        if (price_unit and seller and po.currency_id and
+                seller.currency_id != po.currency_id):
+            price_unit = seller.currency_id.compute(
+                price_unit, po.currency_id)
         vals = po_line_obj.onchange_product_id()
```

Comment:
So, I think the problem is the onchange isn't launched on a memory record.
See https://github.com/odoo/odoo/blob/9ec95b2a682ca885c0796c8e299ac720e8b7e5d8/addons/purchase/models/purchase.py#L726
I fix that in my PR for the migration to 10.0